src/components/MoneyReportHeader.tsx

```diff
@@ -157,6 +158,9 @@ function MoneyReportHeader({policy, report: moneyRequestReport, transactionThrea
     const {isDelegateAccessRestricted, delegatorEmail} = useDelegateUserDetails();
     const [isNoDelegateAccessMenuVisible, setIsNoDelegateAccessMenuVisible] = useState(false);
 
+    const isReportInRHP = isReportOpenInRHP(navigationRef?.getRootState());
+    const shouldDisplaySearchRouter = !isReportInRHP;
```

Comment:
Coming from https://github.com/Expensify/App/issues/51730 where we had the issue that whilst RHP is closing, the last route won't be report RHP anymore, so `isReportInRHP` becomes false and the search icon is briefly shown (as the RHP is closing).